### ISSUE TITLE
loss function missparametrization in vl_simplenn

### ISSUE THREAD

**Issue by sergio-ms** [vlfeat/matconvnet#855]
In the current release (beta23), to create a loss layer you should do like this over your network structure:

net.layers{end+1} = struct('type', 'loss', 'class','log') ;

However, in **examples/cnn_train.m**, line 315, it erases the loss type and replaces it by the labels:

_315: net.layers{end}.class = labels ;_

and then send to **vl_simplenn**, which also have a small problem on lines 329 and 443:

_329: res(i+1).x = vl_nnloss(res(i).x, l.class) ;_
_443: res(i).dzdx = vl_nnloss(res(i).x, l.class, res(i+1).dzdx) ;_

They don't send the loss function type, which is captured by vl_nnloss.m through "varargin" parameters. The correct parametrization should be something like:

_329: res(i+1).x = vl_nnloss(res(i).x, l.labels, [], 'loss', l.class) ;_
_443: res(i).dzdx = vl_nnloss(res(i).x, l.labels, res(i+1).dzdx, 'loss', l.class) ;_

Because the 'loss' parameter is not set, **everytime it computes the "softmaxlog"**, which is the default loss function in vl_nnlosss.
